web/src/components/DandiAppBar.vue

```diff
@@ -81,13 +93,38 @@
 </template>
 
 <script>
-import { mapActions, mapGetters } from 'vuex';
+import { mapActions, mapGetters, mapState } from 'vuex';
 import { Search as GirderSearch, Authentication as GirderAuth } from '@girder/components/src/components';
 
 export default {
   components: { GirderSearch, GirderAuth },
+  asyncComputed: {
+    async apiKey() {
+      const { status, data } = await this.girderRest.get(
+        `api_key?userId=${this.user._id}&limit=50&sort=name&sortdir=1`,
+      );
+      if (status === 200 && data[0]) {
+        // if there is an existing api key
+        if (data[0]._modelType === 'api_key') {
+          // set the user key
+          this.user.key = data[0].key;
+          return this.user.key;
+        }
+      } else {
+        // create a key using "POST" endpoint
+        const { status2, data2 } = await this.girderRest.post(
+          '/api_key?name=dandicli&scope=%20%5B%22core.data.read%22%2C%20%20%22core.data.write%22%5D&tokenDuration=30&active=true',
+        );
+        if (status2 === 200) {
+          this.user.key = data2.key;
+          return this.user.key;
+        }
+      }
+    },
+  },
   computed: {
     ...mapGetters(['loggedIn', 'user']),
+    ...mapState(['girderRest']),
```

Comment:
Then you could mapActions to get your API Key and not import girderRest like this.

One of the main benefits is that when you eventually change to using a different backend, this component requires a lot less work to refactor.